porch/controllers/packagevariantsets/pkg/controllers/packagevariantset/packagevariantset_controller.go

```diff
@@ -76,36 +82,52 @@ func (r *PackageVariantSetReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{}, nil
 	}
 
+	defer func() {
+		r.Client.Status().Update(ctx, pvs)
+	}()
+
 	if errs := validatePackageVariantSet(pvs); len(errs) > 0 {
-		pvs.Status.ValidationErrors = nil
-		for _, validationErr := range errs {
-			if validationErr.Error() != "" {
-				pvs.Status.ValidationErrors = append(pvs.Status.ValidationErrors, validationErr.Error())
-			}
-		}
-		statusUpdateErr := r.Client.Status().Update(ctx, pvs)
-		return ctrl.Result{}, statusUpdateErr
+		validationErr := combineErrors(errs)
+		meta.SetStatusCondition(&pvs.Status.Conditions, metav1.Condition{
+			Type:    "Valid",
+			Status:  "False",
+			Reason:  "Invalid",
+			Message: validationErr,
+		})
+		return ctrl.Result{}, fmt.Errorf(validationErr)
+	} else {
+		meta.SetStatusCondition(&pvs.Status.Conditions, metav1.Condition{
+			Type:    "Valid",
+			Status:  "True",
+			Reason:  "Valid",
+			Message: "all validation checks passed",
+		})
 	}
 
 	upstream, err := r.getUpstreamPR(pvs.Spec.Upstream)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 	if upstream == nil {
+		// Currently, this code will never be reached, because the upstream.Tag option
+		// is not yet implemented.
 		return ctrl.Result{}, fmt.Errorf("could not find specified upstream")
 	}
 
-	downstreams, err := r.unrollDownstreamTargets(ctx, pvs,
-		upstream.Package)
+	downstreams, err := r.unrollDownstreamTargets(ctx, pvs, upstream.Package)
 	if err != nil {
+		if meta.IsNoMatchError(err) {
```

Comment:
I don't think we want to return the error in this situation. It will trigger a new reconcile, which seems unnecessary for these errors.